### PR TITLE
Pin to a single version of chef client consistently in omnibus

### PR DESCRIPTION
**Pull request by PrajaktaPurohit**
A lot of software dependencies pull a floating latest version of the chef-client as a result of which we have multiple versions of clients installed on the chef-server.

This will make it so that a single version of the chef-client is installed on the chef-server.

A better way to do this would be to remove the omnibus-software dependencies for gems and install them from the omnibus/Gemfile (or the respective Gemfiles) #2060 

Signed-off-by: Prajakta Purohit <prajakta@chef.io>

#2033 

https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/1609
https://buildkite.com/chef/chef-chef-server-master-habitat-build/builds/268
https://buildkite.com/chef/chef-chef-server-master-integration-test/builds/322

### CODE REVIEW

File: omnibus/Gemfile [hunk 1]
```diff
@@ -1,11 +1,12 @@
 source 'https://rubygems.org'
 
 gem 'rake'
-
+# chef-cli pulls chef >= 15.0 which is 16.1.16 at the time of failure.
+# https://github.com/chef/chef/commit/003fbc132935961e93667f11c3f45ce4914b83ac
+gem 'chef', '=15.8.23'
```
Comment (tas50) [inline omnibus/Gemfile:6]: You can probably get away with 15.11.whatever here. 16.something is where the change was made

Comment (PrajaktaPurohit) [reply omnibus/Gemfile:6]: Right - I wanted to pin to the same version pinned in omnibus_overrides. We can upgrade the version overall once we know this change works.

File: omnibus/Gemfile [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+# chef-cli pulls chef >= 15.0 which is 16.1.16 at the time of failure.
+# https://github.com/chef/chef/commit/003fbc132935961e93667f11c3f45ce4914b83ac
+gem 'chef', '=15.12.22'
 gem 'chefspec'
 # chef-cli >=3.0.4 requires Ruby version >= 2.7 which chef-server isn't ready for
 gem 'chef-cli', '=3.0.1'
-
 gem 'berkshelf'
-gem 'bundler', '>1.10'
+gem 'bundler', '=1.17.2'
```
Comment (stevendanna) [inline omnibus/Gemfile:11]: Note for other reviewers, our current plan is to try so sort out the bundler version shenanigans once oc-id is updated, as that has been the main pain-point in attempting to upgrade to bundler 2 in the past.

File: omnibus/config/software/fixie-gem.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@
 #
 
 name "chef_fixie"
-default_version "0.4.0"
+default_version "master"
```
Comment (markan) [inline omnibus/config/software/fixie-gem.rb:18]: Yeah, this is probably for the best. It's not like the update rate of fixie is high.

**Review by stevendanna — APPROVED** [review]

**Pull request merged**